TypeScript-tests.md

```diff
@@ -0,0 +1,40 @@
+## Table of contents
+
+* [Introduction](#introduction)
+* [Run TypeScript tests](#run-typescript-tests)
+
+## Introduction
+
+We use a lot of [TypeScript](https://www.typescriptlang.org/) code at Oppia, which lets us write JavaScript code that is strongly typed. We declare the types of variables, function arguments, and function return values. Then, TypeScript checks that there aren't any type errors when it compiles our code. For example, if you have a function that expects an int, TypeScript won't let you call the function with a string.
+
+For help writing TypeScript code, check out our [[guide on defining types|Guide-on-defining-types]]. This page will focus on the TypeScript tests we run to make sure the types are correct.
+
+## Run TypeScript tests
+
+You can run the TypeScript tests like this:
+
+```console
+python -m scripts.typescript_checks
+```
+
+These tests compile all ts files in the codebase and check for errors (including type errors) during compilation. The compiled files are put in the folder `local_compiled_js_for_test`, which is automatically deleted after the tests.
```

Comment:
...after the tests have finished running.

Is it deleted if the test run is cancelled midway?